### ISSUE TITLE
CTkLabel automatically changes width and height when adding new text

### ISSUE THREAD

**Issue by MrBananaPants** [TomSchimansky/CustomTkinter#303]
I've come across a really annoying bug or issue while programming an app.
Whenever I add text to a CTkLabel widget, it increases the width and height of the Label widget even though the width has been set to a fixed value, and the text never exceeds the widget borders.

Here's some code so you can reproduce this issue:
```
import customtkinter as ctk


def add_text():
    text_label.configure(text=text_label.text + str("text") + "\n")


app = ctk.CTk()
app.geometry("500x500")
app.configure(bg="#f2f2f2")

main_frame = ctk.CTkFrame(app, fg_color="#f2f2f2")
main_frame.pack(anchor="w", fill="both", expand=True)

add_text_button = ctk.CTkButton(master=main_frame, fg_color="#3C99DC", width=50, height=25, text="Add text", command=add_text)
add_text_button.place(relx=0.1, rely=0.05, anchor=ctk.CENTER)

text_label = ctk.CTkLabel(master=main_frame, width=80, text="", fg="white", bg_color="#999999", justify="left")
text_label.place(relx=0.1, rely=0.1, anchor=ctk.N)

if __name__ == "__main__":
    app.mainloop()
```
As you can see, whenever you press the button to add some text to the widget, the size increases even though it doesn't need to and I don't want it to.

It's useful that it automatically increases its height so the added text is visible but it clearly has no reason to also increase the width (especially when I've set a fixed width for the widget).

Here's a video of the issue:

![Screen-Recording-2022-07-23-at-15 46 16](https://user-images.githubusercontent.com/41532982/180607910-58262bf6-7e5b-450b-9834-a7f25692a079.gif)

It looks like the widget automatically increases its height so that the newly added text is visible but it wants to keep its aspect ratio so it also has to increase its width. Maybe you could add an option to disable automatic resizing for width and or height or an option to make the aspect ratio not fixed so it only changes the height but not the width (or vice versa).

**Comment by TheFallen-Cat** [comment]
You should remove `expand=True` from your `main_frame`. Im not on pc so i cant run the code, but that must be the issue.

**Comment by felipetesc** [comment]
I tested your code and the defined width wasn't respected by the label, as you have mentioned.

However with some modifications I achieved a different result, by wrapping a single element inside the frame widget, that is, I placed only the label inside a frame. It doesn't solve the issue, but it may be used as a dirty workaround while the definite solution doesn't come out.

```python 
import customtkinter as ctk


def add_text():
    txt =  text_label.text + str("text") + "\n"
    text_label.set_text(text=txt)


app = ctk.CTk()
app.geometry("800x600")
app.configure(bg="blue")

main_frame = ctk.CTkFrame(app, fg_color="red", width=150, height=800)
main_frame.place(relx=0.5, rely=0.5, anchor=ctk.CENTER)



add_text_button = ctk.CTkButton(master=app, fg_color="#3C99DC", width=150, height=25, text="Add text", command=add_text)
add_text_button.place(relx=0.5, rely=0.05, anchor=ctk.CENTER)

text_label = ctk.CTkLabel(master=main_frame, width=80, text="", fg="white", bg_color="#999999", justify="left")
text_label.place(relx=0.5, rely=0.5, anchor=ctk.N)

if __name__ == "__main__":
    app.mainloop()

```
Maybe the code above it can help other people.

![Peek 23-07-2022 13-55](https://user-images.githubusercontent.com/2336812/180615107-3704c32f-9775-474b-b88b-acceab5a1853.gif)

**Comment by felipetesc** [comment]
Maybe the upcomming textbox should be used instead ?

**Comment by TomSchimansky** [comment]
Its a bug in the CTkLabel implementation in line 152, I set the label width to the length of the string, but for multilane strings this doesn't make sense. It will be fixed in the next version.

**Comment by TomSchimansky** [comment]
@MrBananaPants Besides that, I saw you passed an fg argument to the CTkLabel. It worked in this case because the unused arguments get passed to the underlaying tkinter label, but that's not how it should be done. To change the text color use the `text_color` attribute. Here you can see how the attributes are named:
https://github.com/TomSchimansky/CustomTkinter/wiki/Themes

**Comment by MrBananaPants** [comment]
@TomSchimansky Thank you for fixing this bug! Is there a way I can use the master branch in my project already before the next version is released?

About the `fg="white"` argument, it was just a mistake of me because I created the code snippet without too much attention. In my app, I always use `text_color="white"`

**Comment by TomSchimansky** [comment]
I uploaded version 4.5.10 now.

**Comment by MrBananaPants** [comment]
I'm working on a new app now and it looks like this issue still isn't completely fixed in version 4.5.10.
When I set a fixed width for the widget, it still overrides the set width when putting a long string in the label widget even though the length of the string doesn't exceed the width of the label widget. The string fits in the label widget without it needing to resize but it still resizes.
